tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_advertiser.c

```diff
@@ -170,6 +225,106 @@ static void rx_xmit_adv(void)
 	ASSERT_FALSE(err && err != -EALREADY, "stopping scan failed (err %d)", err);
 }
 
+static void send_order_start_cb(uint16_t duration, int err, void *user_data)
+{
+	struct net_buf *buf = (struct net_buf *)user_data;
+
+	ASSERT_OK(err, "Failed adv start cb err (%d)", err);
+	ASSERT_EQUAL(2, buf->len);
+
+	uint8_t current = buf->data[0];
+	uint8_t previous = buf->data[1];
+
+	LOG_INF("tx start: current(%d) previous(%d)", current, previous);
+
+	ASSERT_EQUAL(previous_checker, previous);
+	previous_checker = current;
+}
+
+static void send_order_end_cb(int err, void *user_data)
+{
+	struct net_buf *buf = (struct net_buf *)user_data;
+
+	ASSERT_OK(err, "Failed adv start cb err (%d)", err);
+	ASSERT_TRUE(!buf->data, "Data not cleared!");
+	seq_checker++;
+	LOG_INF("tx end: seq(%d)", seq_checker);
+
+	if (seq_checker == CONFIG_BT_MESH_ADV_BUF_COUNT) {
+		previous_checker = 0xff;
+		k_sem_give(&observer_sem);
+	}
+}
+
+static void receive_order_scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type,
+			struct net_buf_simple *buf)
+{
+	static int cnt;
+	static int64_t buff_timestamp;
+	int64_t timestamp = k_uptime_get();
+	static int64_t delta;
+	uint8_t length;
+	uint8_t current;
+	uint8_t previous;
+
+	length = net_buf_simple_pull_u8(buf);
+	ASSERT_EQUAL(buf->len, length);
+	ASSERT_EQUAL(BT_DATA_MESH_MESSAGE, net_buf_simple_pull_u8(buf));
+	current = net_buf_simple_pull_u8(buf);
+	previous = net_buf_simple_pull_u8(buf);
+	ASSERT_EQUAL(previous_checker, previous);
+
+	cnt++;
+	if (cnt > 1) {
+		delta = k_uptime_delta(&buff_timestamp);
+
+		LOG_INF("rx: cnt(%d) current(%d) previous(%d) delta(%dms)",
+			cnt, current, previous, delta);
+		ASSERT_TRUE(delta >= xmit_param->interval &&
+			delta < (xmit_param->interval + xmit_param->interval));
+	} else {
+		LOG_INF("rx: cnt(%d) current(%d) previous(%d) delta(%dms)",
+			cnt, current, previous, delta);
+	}
+
+	buff_timestamp = timestamp;
+
+	if (cnt > xmit_param->retr) {
+		cnt = 0;
+		delta = 0;
+		buff_timestamp = 0;
+		previous_checker = current;
+		k_sem_give(&observer_sem);
+	}
+}
+
+static void receive_order(int adv_received)
+{
+	struct bt_le_scan_param scan_param = {
+		.type       = BT_HCI_LE_SCAN_PASSIVE,
+		.options    = BT_LE_SCAN_OPT_NONE,
+		.interval   = BT_MESH_ADV_SCAN_UNIT(1000),
+		.window     = BT_MESH_ADV_SCAN_UNIT(1000)
+	};
+	int err;
+
+	k_sem_init(&observer_sem, 0, 1);
+	err = bt_le_scan_start(&scan_param, receive_order_scan_cb);
+	ASSERT_FALSE(err && err != -EALREADY, "starting scan failed (err %d)", err);
+
+	previous_checker = 0xff;
+	for (int i = 0; i < adv_received; i++) {
+		err = k_sem_take(&observer_sem,
```

Comment:
This will be called CONFIG_BT_MESH_ADV_BUF_COUNT for send and reverse. It will be called once for every unique adv message. The timeout will depend on the number of retries, but it need to receive all the retries.

Reply:
This will be called CONFIG_BT_MESH_ADV_BUF_COUNT for send and reverse. It will be called once for every unique adv message. The timeout will depend on the number of retries, but it need to receive all the retries.